src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirJsonInputFormatter.cs

```diff
@@ -56,6 +56,29 @@ public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputForma
 
             var request = context.HttpContext.Request;
 
+            var parserToUse = _parser;
+
+            if (request.Method == "POST"
+               && request.Path.Value.EndsWith("/", System.StringComparison.OrdinalIgnoreCase))
```

Comment:
It might be possible to leverage some of the routing metadata to check if you're deserializing into a bundle, i.e.

```suggestion
            if (string.Equals(request.Method, HttpMethods.Post, StringComparison.OrdinalIgnoreCase) &&
                string.Equals(context.Metadata.Name, KnownActionParameterNames.Bundle, StringComparison.Ordinal))
```